### PR TITLE
Feature/optimize task results calculations

### PR DESCRIPTION
**Pull request by qschmick**
* Added attribute functions to Task model for averageRuntime and lastResult
    * These use the base relationship query to have the database perform action instead of loading entire result dataset into memory
    * Causes better performance and reduces memory issues on dashboard with massive result sets
* Updated index.blade to use new attributes

### CODE REVIEW

File: src/Task.php [hunk 1]
```diff
@@ -97,6 +97,24 @@ public function results()
...
+    {
+        return $this->results()->orderBy('id', 'desc')->first();
+    }
+
+    /**
+     * @return float
+     */
+    public function getAverageRuntimeAttribute()
+    {
+        return $this->results()->avg('duration') ?? 0.00;
```
Comment (roshangautam) [inline src/Task.php:114]: What will happen if results relation is null ?

**Comment by qschmick** [comment]
@roshangautam Were you able to look at this at all?

**Comment by roshangautam** [comment]
Not really. I think it would be a great idea to add a test case around this to make sure nothing breaks if the results relation is null.

**Comment by roshangautam** [comment]
Other than that looks good.

**Comment by qschmick** [comment]
@roshangautam Added tests and a new factory for the Result class.

**Review by roshangautam — APPROVED** [review]

File: src/Task.php (hunk 1 above)
Comment (qschmick) [reply src/Task.php:114]: The null coalesce operator will return 0.00. Additionally, I'm working on the relationship builder and calculating the avg on the database so that could return a null if no records exist which is handled by the operator. There are tests in the PR that cover that possibility with the viewing of the dashboard :)

Comment (robbydooo) [reply src/Task.php:114]: This change has broken my admin on Postgres as would be expected as it is a string field and not numeric field.

  `SQLSTATE[42883]: Undefined function: 7 ERROR: function avg(character varying) does not exist LINE 1: select avg("duration") as aggregate from "task_results" wher... ^ HINT: No function matches the given name and argument types. You might need to add explicit type casts`

from migration table 
     $table->string('duration');

Comment (robbydooo) [reply src/Task.php:114]: This change has broken the admin on Postgres as would be expected as it is a string field and not numeric field.

  `SQLSTATE[42883]: Undefined function: 7 ERROR: function avg(character varying) does not exist LINE 1: select avg("duration") as aggregate from "task_results" wher... ^ HINT: No function matches the given name and argument types. You might need to add explicit type casts`

from migration table 
     $table->string('duration');
